### PR TITLE
[Documentation] updated WSL and Linux document

### PR DESCRIPTION
**Pull request by matthewfcarlson**
We had some feedback that pytools wasn't doing a great job of supporting non windows platforms. We updated the document to include information for correctly installing Mono.

### CODE REVIEW

**Comment by spbrogan** [comment]
@matthewfcarlson  - I think this doc should go away.  Most of this content belongs with a repository that can be more "opinionated" about how things should be setup and work.  Can you please move the critical parts of this doc to https://microsoft.github.io/mu/CodeDevelopment/prerequisites/#windows-subsystem-for-linux-wsl

Then I think we need to update the this doc https://github.com/tianocore/edk2-pytool-extensions/blob/master/docs/usability/using_extdep.md#nuget-dependency with a quick section on nuget/mono on linux based systems.  Really it should just say it doesn't work with native linux distro managed packages and requires the user to follow the directions on the mono-project website.

**Comment by matthewfcarlson** [comment]
@spbrogan Let me know what you think

File: docs/usability/using_extdep.md [hunk 1]
```diff
@@ -51,7 +51,30 @@ Git dependency is used to describe a dependency on a git repository. This reposi
...
+echo "deb https://download.mono-project.com/repo/ubuntu stable-bionic main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list
+sudo apt update
+
+sudo apt upgrade (if mono is already installed)
+- or -
+sudo apt install mono-devel
+```
+
+If you're running a different kind of package manager, or on a system without a package manager, visit the link above for instructions on your platform.
+If you're on a system that NuGet supports (currently only Windows) you won't need to install Mono.
```
Comment (corthon) [inline docs/usability/using_extdep.md]: Natively supports.

File: docs/usability/using_linux.md [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
-If you have a Linux machine, you can follow the later half of this guide to just setup the environment.
+There are many folks in the UEFI community using Linux as their main dev machines.
+This guide is focused on setting up Linux for UEFI development.
+This guide also uses Visual Studio Code as it has fantastic capabilities that makes working in Linux much easier.
+Any IDE or editor is usable for UEFI development.
+It also explains how to use WSL to setup Linux in a Windows environment.
+So if you have a Linux machine, you can follow the later half of this guide to skip setting up WSL.
 
-For reference, this tutorial was written for a 1903 version of Windows.
+For reference, this tutorial was written for a 1903 version of Windows of WSL.
```
Comment (corthon) [inline docs/usability/using_linux.md]: Windows with?

File: docs/usability/using_linux.md [hunk 3]
```diff
@@ -63,31 +64,19 @@ When you first start the environment, it will do initial first time setup that m
...
-Run this command in WSL to install the pieces needed.
-
-```bash
-sudo apt-get install build-essential uuid-dev iasl git gcc-5 nasm python3-distutils
-```
+If you use the external dependency features of pytools, you'll need to update your mono to support NuGet on Linux.
+This applies to all Linux users, WSL or native.
 
-If you aren't use Project Mu's BASECORE, you'll need to compile the BaseTools.
+See more information in the using_extdep document.
```
Comment (corthon) [inline docs/usability/using_linux.md]: Link?

Comment (matthewfcarlson) [reply docs/usability/using_linux.md]: Added

**Pull request merged**